modules/subscriptions.php

```diff
@@ -616,7 +616,7 @@ function widget( $args, $instance ) {
 					<p class="error"><?php esc_html_e( 'You have already subscribed to this site. Please check your inbox.', 'jetpack' ); ?></p>
 				<?php break;
 				case 'success' :
-					echo wpautop( $success_message );
+					echo wpautop( str_replace( '%d', number_format_i18n( $subscribers_total['value'] ), $success_message ) );
```

Comment:
This looks good.  The only issue I have is that the success message isn't style able.  It would be really nice if you could pass a class to wpautop, but, alas.

Maybe wrap it in a div with a "jetpack-subscribe-success" class?